Source/Python/WarpXWrappers.cpp

```diff
@@ -421,6 +421,33 @@ extern "C"
         return data;
     }
 
+    amrex::ParticleReal** warpx_getParticleArraysFromCompName (
+            int speciesnumber, const char* char_comp_name,
+            int lev, int* num_tiles, int** particles_per_tile ) {
+
+        const auto & mypc = WarpX::GetInstance().GetPartContainer();
+        auto & myspc = mypc.GetParticleContainer(speciesnumber);
+
+        const std::string comp_name(char_comp_name);
+        auto particle_comps = myspc.getParticleComps();
+        auto comp = particle_comps.at(comp_name);
+
+        return warpx_getParticleArrays(
+            speciesnumber, comp, lev, num_tiles, particles_per_tile
+        );
+    }
+
+    void warpx_addRealComp(const char* char_comp_name, bool comm=true)
```

Comment:
Generally, when we add extra components at runtime, we only want to do it for certain particle species. Should this function (or maybe another version of this function) take the species as well, to expose the ability to only add the component for certain species from python?